File: Backend/src/main/kotlin/dev/kocken/backend/models/Event.kt
package dev.kocken.backend.models

import com.google.cloud.Timestamp
import dev.kocken.backend.models.firebase.FirebaseEvent
import java.time.LocalDateTime

data class Event(
    val name: String,
    val timestamp: LocalDateTime,
    val value: Number,
)

fun Event.toFirebase(
    name: String, timestamp: LocalDateTime, value: Number
): FirebaseEvent {
    return FirebaseEvent(name, Timestamp.parseTimestamp(timestamp.toString()), value.toString())
}

File: XR/Assets/Scripts/Heart/Event.cs
using System;
using UnityEngine;

namespace Api
{
    public class Event
    {
        public string Name { get; }

        public DateTime Timestamp { get; }

        public int Value { get; }

        public Event(string name, int value)
        {
            Name = name;
            Timestamp = DateTime.Now;
            Value = value;
        }
    }
}

